activerecord/lib/active_record/associations/has_many_association.rb

```diff
@@ -95,7 +95,14 @@ def delete_count(method, scope)
           if method == :delete_all
             scope.delete_all
           else
-            scope.update_all(reflection.foreign_key => nil)
+            scope.update_all(columns_to_nullify)
+          end
+        end
+
+        def columns_to_nullify
+          Hash.new.tap do |h|
```

Comment:
I think using `h` as a variable name isn't very descriptive. Maybe `columns` or `attributes` is better?